clang-tools-extra/docs/clang-tidy/index.rst

```diff
@@ -41,6 +41,38 @@ parameters file:
 
   $ clang-tidy @parameters_file
 
+Running Clang-Tidy on Multiple Files
+=====================================
+
+For projects with multiple source files, you can use the :program:`run-clang-tidy.py`
+script to run :program:`clang-tidy` over all files in a compilation database in
+parallel. This script is included with clang-tidy and provides a convenient way
+to analyze entire projects efficiently.
+
+The script requires a compilation database (``compile_commands.json``) which can
+be generated by build systems like CMake (using ``-DCMAKE_EXPORT_COMPILE_COMMANDS=ON``)
+or by tools like Bear.
+
+Example invocations:
+
+.. code-block:: console
+
+  # Run clang-tidy on all files in the compilation database
+  $ run-clang-tidy.py
+
+  # Run with specific checks and apply fixes
+  $ run-clang-tidy.py -fix -checks=-*,readability-*
+
+  # Run on specific files/directories with header filtering
+  $ run-clang-tidy.py src/ -header-filter=src/
+
+  # Run with parallel execution (uses all CPU cores by default)
+  $ run-clang-tidy.py -j 4
+
+The script supports most of the same options as :program:`clang-tidy` itself,
+including ``-checks=``, ``-fix``, ``-header-filter=``, and configuration options.
+Run ``run-clang-tidy.py --help`` for a complete list of available options.
+
```

Comment:
The block of text below is not related to this section, so a new header should be created for it (or this section moved elsewhere).